packages/firestore/src/api/timestamp.ts

```diff
@@ -83,6 +83,14 @@ export class Timestamp {
     );
   }
 
+  _toISOString(): string {
+    const strMillis = this.toDate().toISOString(); // JS Date only has millis.
+    // So let's replace the fraction w/ up to 9 digits (nano resolution).
+    const fracs = (this.nanoseconds / 1e9).toFixed(9).substr(1); // .123456789
+    const strNanos = strMillis.replace(/\.\d*/, '').replace('Z', fracs + 'Z');
+    return strNanos;
```

Comment:
Looks like my comment got dropped. I think we should try to make it more obvious what is going on, so that a casual read can understand this.

My idea would be:

- Create a Date just from the seconds
- Drop the last 4 characters from its toISOString
- Zero pad the nanos by prefixing with "000000000" and using substring
- Concat both strings and add "Z"

Reply:
Done